### ISSUE TITLE
Mod completely breaks on some servers.

### ISSUE THREAD

**Issue by IceBlossom** [TerraformersMC/Terrestria#332]
I couldn't really think of a better title, but these screenshots should do justice.
I have confirmed this happens when JUST Terrestria (+Fabric API) is installed.
![image](https://github.com/user-attachments/assets/ea8eecb5-be1c-43d7-b248-ead0cce581c8)
![image](https://github.com/user-attachments/assets/9dab270c-95bd-4749-9bf0-a2f9a791d2e8)
First image is on MCC Island, second is on Hypixel.

**Comment by gniftygnome** [comment]
Are you running a vanilla client without Terrestria installed?  That does not work and is not intended to work.

**Comment by IceBlossom** [comment]
No. This is on Fabric (Loader version 0.16.5) and the latest version of Fabric API for 1.21.1

**Comment by gniftygnome** [comment]
Please provide the client logs.

**Comment by gniftygnome** [comment]
Actually, when you say it breaks on some servers, do the servers have Terrestria installed?  It must be installed on both and it must be the same version.

**Comment by IceBlossom** [comment]
No. I don't wanna try and sound rude, but why would practically any non-survival server with vanilla compatibility have world gen mods installed? 
The logs say practically nothing, but here you go.
[latest.log](https://github.com/user-attachments/files/17267765/latest.log)

**Comment by gniftygnome** [comment]
Mods that add blocks and items to the game have to be on the server.  That is a fundamental requirement.  Some use clever hacks to avoid having to be installed on the client (but Terrestria is not one of these).  Having Terrestria installed on the client and not on the server is not a supported configuration.

**Comment by IceBlossom** [comment]
I understand that having the mod not on the client side wouldn't allow it to be used. However, wouldn't the client just not be able to use it considering the server doesn't have any of the relevant data? And so nothing would be wrong?
If that's the case, then why does having the mod installed on the client break _everything_ on a server?
These screenshots, like the ones in the start of this thread, are on MCC Island, Hypixel, and Minr's lobbies with JUST Terrestria and Fabric API/Loader.
![2024-10-05_22 01 41](https://github.com/user-attachments/assets/4d7b32d0-f775-4d95-b620-a7552f92f54f)
![2024-10-05_22 01 56](https://github.com/user-attachments/assets/00225ef5-15a2-4f34-8976-2eab2e5aeb4e)
![2024-10-05_22 02 06](https://github.com/user-attachments/assets/ebbb9b96-e420-48a1-88c1-9b45ff94d0e3)

And by the way, all those blocks. Ones that aren't there in the server, all the waterlogged blocks. They all act like they are actually there on the client, so I am able to stand on those. To other players, it looks like I'm floating. 


![2024-10-05_22 16 42](https://github.com/user-attachments/assets/a7633474-7495-487c-b6a4-58b9c3f87d87)

![names](https://github.com/user-attachments/assets/fa559093-ab9f-466e-a92a-73bf1982655d)

When looking at the F3 screen, looking at the blocks that appear on the client side do say what they are, even when in reality it should be air, or a completely different block.



These screenshots were taken on my normal modpack with 100+ mods without Terrestria (including world gen mods like Structory, Terralith, Tectonic, and Blooming Biosphere for singleplayer use, which is what I installed Terrestria for) and does not affect what I see on vanilla servers at all (can't include Minr due to it not telling me what mod I have that's problematic).

![2024-10-05_22 07 28](https://github.com/user-attachments/assets/a7c6abc8-3350-4214-a6b4-701d06867500)
![2024-10-05_22 07 45](https://github.com/user-attachments/assets/1cda87f3-0010-49f7-b936-9602ee0c2fd1)

Why Terrestria would cause this huge of an issue on servers that simply don't have the mod installed is beyond me.

**Comment by gniftygnome** [comment]
It doesn't break anything, the texture locations just aren't where they are on the server, so the client shows you the wrong textures.

**Comment by IceBlossom** [comment]
Okay, so I could get it showing me the wrong textures. But why would the mod put in entirely new blocks that are no where to be found on a vanilla client? And change the properties of some blocks to be waterlogged? Incorrect textures should not be able to do that.

**Comment by gniftygnome** [comment]
They should just be waterlogged on the client.  Block states are part of how textures are selected, so selecting the wrong texture can mean selecting a waterlogged one.  Same with seeing the modded blocks.  Nothing changed on the server, but on the client you can see a modded block because those modded blocks are now part of the client's blocks and texture atlas.

**Comment by 8030s** [comment]
I got the same problem on my local server. I just made a Fabric server, and it doesn't work. It works properly on singleplayer, but it doesn't work on the server.

**Comment by gniftygnome** [comment]
> It works properly on singleplayer, but it doesn't work on the server.

If you have correctly installed Terrestria on both the client and server, then this is a separate issue and you should file a new one and include the logs of both the client and server.

If you are attempting to use Terrestria only on the client or only on the server, then this is expected behavior and it is unlikely to be resolved any time soon, as Minecraft's design decisions w.r.t. leaves blocks makes it nearly impossible to do so without breaking compatibility with a whole bunch of other mods.

**Comment by gniftygnome** [comment]
This was resolved in Terraform API for 1.21.5, by using a super sketchy hack to evaluate leaf support using both the vanilla distance property and an auxiliary extended distance property.  Unfortunately, this has poor compatibility with leaf decay mods, but it should eventually stop the complaints from people who think it makes sense to run a biome mod on the client but not the server.
